frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -304,22 +310,26 @@ public void shutdown() {
     }
 
     void setState(WorkerState newState) {
+        logger.debug("State {} -> {}", state, newState);
         listener.notifyChangeState(model.getModelName(), newState);
         long timeTaken = System.currentTimeMillis() - startTime;
-        if (state != WorkerState.WORKER_TERMINATED) {
+        if ((state != WorkerState.WORKER_SCALED_DOWN)
+                || (newState == WorkerState.WORKER_SCALED_DOWN)) {
+            // Don't update the state if it was terminated on purpose.. Scaling in..
             this.state = newState;
         }
-
         if (state == WorkerState.WORKER_MODEL_LOADED) {
             workerLoadTime.setValue(String.valueOf(timeTaken));
             workerLoadTime.setTimestamp(
                     String.valueOf(TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis())));
             loggerMmsMetrics.info(workerLoadTime);
         }
+        logger.debug("State changed to {}", state);
```

Comment:
This log is redundant:
1. you have a log at the beginning already
2. there is a log in listener.notifyChangeState() already

Reply:
I don't think we capture state transition .. We only capture what it is set to, but its hard to track what it was before.